firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -176,20 +181,23 @@ public InstallationResponse createFirebaseInstallation(
   private void writeFIDCreateRequestBodyToOutputStream(
       HttpURLConnection httpURLConnection, @NonNull String fid, @NonNull String appId)
       throws IOException {
-    OutputStream outputStream = httpURLConnection.getOutputStream();
-    if (outputStream == null) {
-      throw new IOException(
-          "Cannot send CreateInstallation request to FIS. No OutputStream available.");
-    }
-
-    GZIPOutputStream gzipOutputStream = new GZIPOutputStream(outputStream);
     try {
-      gzipOutputStream.write(
-          buildCreateFirebaseInstallationRequestBody(fid, appId).toString().getBytes("UTF-8"));
+      writeRequestBodyToOutputStream(
+          httpURLConnection, buildCreateFirebaseInstallationRequestBody(fid, appId));
     } catch (JSONException e) {
       throw new IllegalStateException(e);
-    } finally {
-      gzipOutputStream.close();
+    }
+  }
+
+  private static void writeRequestBodyToOutputStream(
+      HttpURLConnection httpURLConnection, JSONObject jsonObject) throws IOException {
+    try (OutputStream outputStream = httpURLConnection.getOutputStream()) {
```

Comment:
I'm not too familiar with this pattern.
Are OutputStreams always AutoCloseable?

Reply:
That's right. OutputStream implements Closeable